packages/components/src/font-size-picker/index.js

```diff
@@ -47,23 +47,41 @@ function FontSizePicker(
 		availableUnits: [ 'px', 'em', 'rem' ],
 	} );
 
-	// The main font size UI displays a toggle group when the presets are less
-	// than six and a select control when they are more.
-	//
-	// A select control is also used when the value of a preset cannot be
-	// immediately computed (eg. 'calc', 'var').
+	/**
+	 * The main font size UI displays a toggle group when the presets are less
+	 * than six and a select control when they are more.
+	 *
+	 * A select control is also used when the value of a preset cannot be
+	 * immediately computed (eg. 'calc', 'var') and there is no `alias` provided
+	 * for every font size.
+	 */
+	const fontSizesContainComplexValues = fontSizes.some(
+		( { size } ) => ! isSimpleCssValue( size )
+	);
+	const allFontSizesHaveAliases = fontSizes.every(
+		( { alias } ) => !! alias
+	);
 	const shouldUseSelectControl =
 		fontSizes.length > 5 ||
-		fontSizes.some( ( { size } ) => ! isSimpleCssValue( size ) );
+		( fontSizesContainComplexValues && ! allFontSizesHaveAliases );
+	const shouldUseAliases =
+		fontSizesContainComplexValues && allFontSizesHaveAliases;
```

Comment:
What happens if a font size is a "simple" value, but has an associated alias? Should the alias take the precedence?

Reply:
Based on the initial design with the intention of showing numeric values, I'd say no..